default-fields/0000-default-fields.md

```diff
@@ -6,31 +6,206 @@
 # Summary
 [summary]: #summary
 
-One para explanation of the feature.
+Allow struct definitions to supply default values for individual fields,
+and then allow those fields to be omitted from struct initialisation:
+
+```rust
+struct Foo {
+    a: &'static str,
+    b: bool = true,
+    c: i32,
+    d: Vec<i32> = Vec::default()
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42,
+}
+```
 
 # Motivation
 [motivation]: #motivation
 
-Why are we doing this? What use cases does it support? What is the expected outcome?
+Today, Rust allows you to create an instance of a struct using a literal syntax.
+This requires all fields in the struct be assigned a value, and can't be used
+with private fields.
+It can also be inconvenient for large structs whose fields usually receive the same values.
+
+With the `..` syntax, values for missing fields can be taken from another struct.
+For example, using `Default` to override just a few values on initialisation:
+
+```rust
+#[derive(Default)]
+struct Foo {
+    a: &'static str,
+    b: bool,
+    c: i32,
+    d: Vec<i32>
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42,
+    ..Default::default()
+}
+```
+
+However, this still requires an already initialised struct after the `..`.
+It also isn't valid if the struct has private fields.
+
+To work around these shortcomings, users can create constructor functions or more elaborate builders.
+The problem with a constructor is that you either need one for each combination of fields a caller can supply
+or expect callers to override values after initialisation, which is inefficient.
+Builders enable more advanced initialisation, but need additional boilerplate.
 
 # Detailed design
 [design]: #detailed-design
 
-This is the bulk of the RFC. Explain the design in enough detail for somebody familiar
-with the language to understand, and for somebody familiar with the compiler to implement.
-This should get into specifics and corner-cases, and include examples of how the feature is used.
+With field defaults a caller can initialise a struct with default values without needing builders
+or a constructor function:
```

Comment:
Good idea. I'll spend some time with [the grammar spec](https://doc.rust-lang.org/grammar.html) and put together a formal spec. This might surface some inconsistencies too.